game/world/objects/npc.cpp

```diff
@@ -1358,16 +1358,20 @@ bool Npc::implGoTo(uint64_t dt, float destDist) {
   else if(mvAlgo.isClose(go2.target(),destDist)) {
     bool finished = true;
     if(go2.flag==GT_Way) {
-      go2.wp = wayPath.pop();
+      go2.wp = go2.wp->hasLadderConn(wayPath.first()) ? wayPath.first() : wayPath.pop();
       if(go2.wp!=nullptr) {
         attachToPoint(go2.wp);
+        if(useLadder()) {
+          mvAlgo.tick(dt);
+          return true;
+          }
         finished = false;
         }
       }
     if(finished)
       clearGoTo();
     } else {
-    if(mvAlgo.checkLastBounce() && implTurnTo(dpos.x,dpos.z,false,dt)){
+    if(useLadder() || (mvAlgo.checkLastBounce() && implTurnTo(dpos.x,dpos.z,false,dt))) {
```

Comment:
Expression is too complex with multiple side-effects inside if.
Can you please split it into 2?
```
if(ladder()) {
  ...
  }
if(checkLastBounce && implTurnTo()) {
  ...
  }
```